streams/src/main/resources/common/message/SubscriptionInfoData.json

```diff
@@ -57,6 +57,11 @@
       "name": "uniqueField",
       "versions": "8+",
       "type": "int8"
+    },
+    {
+      "name": "shutdownRequested",
+      "versions": "9+",
+      "type": "int8"
```

Comment:
I think we should mirror the `errorCode` in the AssignmentInfo here, both in terms of naming and type. If we're going to use the same AssignorError for both, then they should really be the same. And we may want to send other kinds of error codes in the subscription going forward: better to just encode a single `int` than a separate `byte` for every logical error code. I don't think we'll notice the extra three bytes since Subscriptions aren't sent that frequently

Reply:
I think I agree on the name, I am not sure about the type. We should be able to fit thousands of different error code into the byte so we should not run out of space. The reason the errorCode. is an integer in the first place is because there is not `AtomicByte` that I know of.

Reply:
I'm not really worried that we'd run out of space, I just think it sends a signal that the Assignment and Subscription error codes are semantically distinct and don't refer to the same underlying concept. So it seems better to go with the simpler approach than over-optimize to save an occasional three bytes

Reply:
https://github.com/apache/kafka/pull/9273#discussion_r486597512
I originally had it at int32, but john suggested int16, now it is int8.

would you be good with int16 or do you think int32 is the way?